OWML.ModHelper.Events/TypeExtensions.cs

```diff
@@ -67,5 +67,11 @@ public static void Invoke(this object obj, string name, params object[] paramete
             method?.Invoke(obj, parameters);
         }
 
+        public static T Invoke<T>(this object obj, string name, params object[] parameters)
+        {
+            var type = obj.GetType();
+            var method = type.GetAnyMethod(name);
+            return (T)method?.Invoke(obj, parameters);
```

Comment:
I think this can be shortened by calling the other Invoke, then casting:
return (T)obj.Invoke(name, parameters);

Reply:
but that other invoke doesn't return anything

Reply:
oh wait, i'm confused

Reply:
You're right! Then you can do the opposite, call your new method from the original one (without a return) :)

Reply:
Scratch that, let's not mess with existing functionality, just in case.